libp2p/network/connection/raw_connection.py

```diff
@@ -1,42 +1,26 @@
-import asyncio
+import trio
+
+from libp2p.io.abc import ReadWriteCloser
+from libp2p.io.exceptions import IOException
 
 from .exceptions import RawConnError
 from .raw_connection_interface import IRawConnection
 
 
 class RawConnection(IRawConnection):
-    reader: asyncio.StreamReader
-    writer: asyncio.StreamWriter
+    read_write_closer: ReadWriteCloser
     is_initiator: bool
 
-    _drain_lock: asyncio.Lock
-
-    def __init__(
-        self,
-        reader: asyncio.StreamReader,
-        writer: asyncio.StreamWriter,
-        initiator: bool,
-    ) -> None:
-        self.reader = reader
-        self.writer = writer
+    def __init__(self, read_write_closer: ReadWriteCloser, initiator: bool) -> None:
+        self.read_write_closer = read_write_closer
         self.is_initiator = initiator
 
-        self._drain_lock = asyncio.Lock()
-
     async def write(self, data: bytes) -> None:
         """Raise `RawConnError` if the underlying connection breaks."""
         try:
-            self.writer.write(data)
-        except ConnectionResetError as error:
+            await self.read_write_closer.write(data)
```

Comment:
maybe rather than `read_write_closer` this would be better just called `stream`?